doc/STYLE.md

```diff
@@ -0,0 +1,160 @@
+# Care And Feeding of Your Fellow Coders
+
+Style is an individualistic thing, but working on software is group
+activity, so consistency is important.
+
+## Communication
+
+We communicate with each other via code; we polish each others code,
+and give nuanced feedback.  Exceptions to the rules below always
+exist: accept them.  Particularly if they're funny!
+
+## Prefer Short Names
+
+`num_foos` is better than `number_of_foos`, and `i` is better than
+`counter`.  But `bool found;` is better than `bool ret;`.  Be as
+short as you can but still descriptive.
+
+## Prefer 80 Columns
+
+We have to stop somewhere.  The two tools here are extracting
+deeply-indented code into their own functions, and use of short-cuts
+using early returns or continues, eg:
+
+```C
+for (i = start; i != end; i++) {
+	if (i->something)
+		continue;
+
+    if (!i->something_else)
+		continue;
+
+	do_something(i);
+}
+```
+
+## Prefer Simple Statements
+
+Notice the statement above uses separate tests, rather than combining
+them.  I prefer to only combine conditionals which are fundamentally
```

Comment:
"I" as in Rusty -> "we" the c-lightning devs?

Reply:
Didn't want to speak for everyone, but will change since I take comments here as broadly approving :)